tests/oc_template_test.go

```diff
@@ -0,0 +1,257 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+// template parameters
+const (
+	vmName     = "testvm"
+	vmCpuCores = "2"
+)
+
+// parameter strings
+const (
+	vmTemplateParams = "-p NAME=" + vmName + " -p CPU_CORES=" + vmCpuCores
+	vmStartPatch     = "-p '{\"spec\":{\"running\":true}}'"
+	vmStopPatch      = "-p '{\"spec\":{\"running\":false}}'"
+)
+
+var _ = Describe("VM Template", func() {
+	flag.Parse()
+
+	vmTemplate := `{
+		"kind": "Template",
+		"objects": [
+			{
+				"kind": "OfflineVirtualMachine",
```

Comment:
At least the whole `OfflineVirtualMachine` acn be created via our structs. Why can't we use the Template struct like agreed on before? If we don't test mappings between structs and manifests themselves, then raw jsons or yamls are not an option. That leads to harder maintainable code.

Reply:
@rmohr I agree. Nevertheless, the `types.go` file smuggled from the `openshift/origin` contained dependencies that were not met within `kubevirt/kubevirt`. @stu-gott suggested not bringing any KubeVirt-unrelated dependencies into the project, so I have decided to ephemerally drop this file and define Json code by the hardcoded string until I do not figure out how to resolve this properly. Note that this test will need refactoring soon anyway once offlinevirtualmachines will be renamed. Creating Json object by serialization from the template `struct` is vital in the long term. At this very moment, we need test results.